.travis.yml

```diff
@@ -30,8 +30,8 @@ before_cache:
 
 matrix:
   include:
-    - compiler: "ghc-8.6.4"
-      addons: {apt: {packages: [ghc-ppa-tools,cabal-install-2.4,ghc-8.6.4], sources: [hvr-ghc]}}
+    - compiler: "ghc-8.8.1"
+      addons: {apt: {packages: [ghc-ppa-tools,cabal-install-2.4,ghc-8.8.1], sources: [hvr-ghc]}}
```

Comment:
Don't edit the `.travis.yml` file directly. Instead, let's autogenerate this with [`haskell-ci`](https://github.com/haskell-CI/haskell-ci). It should suffice to do the following:

1. Build `haskell-ci` from source at [its GitHub repo](https://github.com/haskell-CI/haskell-ci)
2. Change the `tested-with` stanza in `singletons.cabal` to be `8.8.1`
3. Run `haskell-ci regenerate` in the same directory as `singletons.cabal`

This will also fix the version bounds problem you observed in https://github.com/goldfirere/singletons/pull/390#issuecomment-487273812.

Reply:
Thanks, I'll do that.

Reply:
The 8.8 job was added to `allow_failures` by the script – do I fix it up manually?

Reply:
I vote to keep it the way it is. A number of `singletons`' dependencies only build at the moment since they have `head.hackage` patches, but if a newer version of a dependency were to be released and `head.hackage` lagged behind the latest Hackage version, then it's possible that the build could start to break if Travis picked a build plan that didn't include the patched version.

`haskell-ci` usually switches off the `GHCHEAD` behavior of a pre-release version of GHC once it has been released in full, so we can revisit this once that occurs.

Reply:
Hm, that's not the answer I expected, ha-ha. My question was more along the lines of "do I fix it up manually or via some config option". Not sure if it's a good idea to have all jobs in `allow_failures`, it will be hard to notice if a commit breaks something.

Let's keep it removed until we start seeing spurious failures, I'd rather deal with a false positive than a false negative.

Reply:
Very well. If you're going to change the generated `.travis.yml`, then please add a `travis.yml.patch` file containing that change and put `travis-patches: travis.yml.patch` in `cabal.haskell-ci`. That way, this _post facto_ change will always be applied automatically instead of having to remember to do it each time the script is regenerated. (Since we have to regenerate the script every time the `th-desugar` submodule is updated, this occurs fairly regularly.)

As a reference, here is the [`travis.yml.patch` file](https://github.com/haskell-compat/base-compat/blob/c009c983308bcefb2ca026c81db8ddfd19ca4df6/travis.yml.patch) for `base-compat`.

Reply:
Thanks for the suggestion, I added a `travis.yml.patch` file.